### PR TITLE
[MINOR] Python Lineage Trace Test Change

### PR DESCRIPTION
**Pull request by Baunsgaard**
[MINOR] Python Lineage Trace Test Change
    
Change to the way lineage trace tests are executed, such that instead of
having files to compare to, the python trace is compared to a trace made
from systemds directly.
    
Motivated by the fact that the previous tests were failing, because of
inconsistencies between new traces and old.

### CODE REVIEW

**Comment by Baunsgaard** [comment]
Currently failing because python interface call operations with 1 thread only, while sysds calls with max number of threads.

**Comment by Baunsgaard** [comment]
@phaniarnab comments, please?

File: src/main/python/tests/lineage/test_lineagetrace.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    def tearDown(self):
+        warnings.filterwarnings(
+            action="ignore", message="unclosed", category=ResourceWarning
+        )
+
+    def test_compare_trace1(self):  # test getLineageTrace() on an intermediate
+        m = sds.full((10, 10), 1)
+        m_res = m + m
+
+        python_trace = [x.strip() for x in m_res.get_lineage_trace().split("\n")]
```
Comment (Baunsgaard) [inline src/main/python/tests/lineage/test_lineagetrace.py:51]: Do we really want the trace to be returned as one large string as default, or something like a list?

Comment (phaniarnab) [reply src/main/python/tests/lineage/test_lineagetrace.py:53]: I guess that depends on what one wants to do with the trace. String is ubiquitous, woks in dml and python. Using other data structures might need different handling. But can be done if that helps the querying interface on top the lineage storage.

**Comment by phaniarnab** [comment]
This looks very good. Thanks for making the test better. As discussed offline, comparing only the internal nodes might solve the problem of metadata mismatch.

**Comment by Baunsgaard** [comment]
LGTM :grinning: 
@mboehm7 or @j143 , could you merge ? (It would fix the few issues we have on master now.)

**Comment by mboehm7** [comment]
LGTM - thanks @Baunsgaard. In general this is fine, but in the future, it would be good to preserve the history for renamed/moved files (`src/main/python/tests/test_lineagetrace.py` to
`src/main/python/tests/lineage/test_lineagetrace.py`). What usually works is to first rename it, commit, and make the necessary changes in a subsequent commit.

**Comment by j143** [comment]
Normally, the `refactor` option in the IDEs should preserve git history.

**Comment by mboehm7** [comment]
Yes, git is able to detect plain refactorings - the problem usually arises, if a commit does refactoring AND makes major code changes.

**Comment by Baunsgaard** [comment]
I will note it down, and try to make it so next time :)

**Pull request closed without merging**